packages/block-library/src/image/index.php

```diff
@@ -71,6 +64,202 @@ function render_block_core_image( $attributes, $content, $block ) {
+	// and Chrome (see https://github.com/WordPress/gutenberg/pull/52765#issuecomment-1674008151). Until that
+	// is resolved, manually setting the 'src' seems to be the best solution to load the large image on demand.
+	$q->set_attribute( 'src', '' );
+	$q->set_attribute( 'data-wp-bind--src', 'selectors.core.image.enlargedImgSrc' );
+	$q->set_attribute( 'data-wp-style--object-fit', 'selectors.core.image.lightboxObjectFit' );
+	$enlarged_image_content = $q->get_updated_html();
+
+	$background_color = esc_attr( wp_get_global_styles( array( 'color', 'background' ) ) );
+
+	$close_button_icon  = '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="15" height="15" aria-hidden="true" focusable="false"><path d="M13 11.8l6.1-6.3-1-1-6.1 6.2-6.1-6.2-1 1 6.1 6.3-6.5 6.7 1 1 6.5-6.6 6.5 6.6 1-1z"></path></svg>';
+	$close_button_color = esc_attr( wp_get_global_styles( array( 'color', 'text' ) ) );
+	$dialog_label       = $alt_attribute ? esc_attr( $alt_attribute ) : esc_attr__( 'Image', 'gutenberg' );
+	$close_button_label = esc_attr__( 'Close', 'gutenberg' );
+
+	$lightbox_html = <<<HTML
+        <div data-wp-body="" class="wp-lightbox-overlay $lightbox_animation"
+            data-wp-bind--role="selectors.core.image.roleAttribute"
+            aria-label="$dialog_label"
+            data-wp-class--initialized="context.core.image.initialized"
+            data-wp-class--active="context.core.image.lightboxEnabled"
+			      data-wp-class--hideAnimationEnabled="context.core.image.hideAnimationEnabled"
+            data-wp-bind--aria-hidden="!context.core.image.lightboxEnabled"
+            data-wp-bind--aria-modal="context.core.image.lightboxEnabled"
+            data-wp-effect="effects.core.image.initLightbox"
+            data-wp-on--keydown="actions.core.image.handleKeydown"
+            data-wp-on--mousewheel="actions.core.image.hideLightbox"
+            data-wp-on--click="actions.core.image.hideLightbox"
+            >
+                <button type="button" aria-label="$close_button_label" style="fill: $close_button_color" class="close-button" data-wp-on--click="actions.core.image.hideLightbox">
+                    $close_button_icon
+                </button>
+                <div class="lightbox-image-container">$initial_image_content</div>
+				<div class="lightbox-image-container">$enlarged_image_content</div>
+                <div class="scrim" style="background-color: $background_color"></div>
+        </div>
+HTML;
+
+	return str_replace( '</figure>', $lightbox_html . '</figure>', $body_content );
+}
+
+// TODO: We should not be adding a separate filter but rather move the
+// the lightbox rendering to the `render_block_core_image` function.
+
+// Use priority 15 to run this hook after other hooks/plugins.
+// They could use the `render_block_{$this->name}` filter to modify the markup.
+add_filter( 'render_block_core/image', 'block_core_image_render_lightbox', 15, 2 );
```

Comment:
Noting the TODO for the subsequent PR that adds the revised Lightbox implementation.